src/hotspot/share/runtime/globals.hpp

```diff
@@ -1304,6 +1304,12 @@ const int ObjectAlignmentInBytes = 8;
   develop(int, MaxElementPrintSize, 256,                                    \
           "maximum number of elements to print")                            \
                                                                             \
+  develop(int, MaxStringPrintSize, 256,                                     \
+          "maximum number of characters to print for a java.lang.String "   \
+          "in the VM. If exceeded, an abridged version of the string is "   \
+          "printed with the middle of the string elided.")                  \
+          range(2, O_BUFLEN)                                                \
+                                                                            \
```

Comment:
This would make sense as a product diagnostic switch. You want to be able to increase this at a customer if needed.

Reply:
This is modelled after the `MaxElementPrintSize` that precedes it.